### PR TITLE
Add ability to set http request timeouts in HttpManager

### PR DESCRIPTION
**Pull request by sprokopets**
I created PR in case anyone needs to set custom timeout for http client that can prevent potential issues with 'infinite http requests' (that I have observed).

Also I noticed one potential problem with SpotifyHttpManager but decided not to change code because I am not aware of intentions why 'CloseableHttpClient httpClient' is a static field. At least for me it caused troubles when order of initialization was different for 'custom http client' and 'default http client'.
In other words this looks like a side side effect that can take place when 'custom httpClient' is initialized first, then it can be overridden by other instances of SpotifyHttpManager class e.g. created by 'new SpotifyApi.Builder()'.

### CODE REVIEW

**Comment by Dargmuesli** [comment]
Thank you very much for this well explained addition! I'll push a new release shortly.
Regarding `httpClient`: I'm pretty sure that it was meant to say `final` and not `static`. I corrected that.
